src/components/DatePickerRange.react.js

```diff
@@ -263,6 +264,31 @@ DatePickerRange.propTypes = {
     persistence_type: PropTypes.oneOf(['local', 'session', 'memory']),
 };
 
+DatePickerRange.persistenceTransforms = {
+    end_date: {
+        extract: propValue => {
+            if (!(propValue === null || propValue === undefined)) {
+                return moment(propValue)
+                    .startOf('day')
+                    .format('YYYY-MM-DD');
+            }
+            return propValue;
+        },
+        apply: storedValue => storedValue,
+    },
+    start_date: {
+        extract: propValue => {
+            if (!(propValue === null || propValue === undefined)) {
+                return moment(propValue)
+                    .startOf('day')
+                    .format('YYYY-MM-DD');
+            }
+            return propValue;
+        },
+        apply: storedValue => storedValue,
```

Comment:
The implementation for `end_date` and `start_date` extract and apply are identical (same for DatePickerSingle `date`) -- DRY by defining a functions that you can share between the 3 cases

Reply:
32852c0